TrailBlaze/Objects/Interception.m

```diff
@@ -0,0 +1,54 @@
+//
+//  Interception.m
+//  TrailBlaze
+//
+//  Created by Adam Issah on 7/26/22.
+//
+
+#import "Interception.h"
+#import "MapKit/Mapkit.h"
+@import Parse;
+
+
+@implementation Interception
+@dynamic rendezvous;
+@dynamic polylineCoords;
+@dynamic startTime;
+@dynamic receiver;
+
++ (nonnull NSString *)parseClassName {
+    return @"Interception";
+}
+
++ (void) uploadRequest: (PFGeoPoint *) rendezvous polyline: (MKPolyline *)polyline receiver: (PFUser *) receiver withCompletion: (PFBooleanResultBlock _Nullable)completion {
+    
+    Interception *newReq = [Interception new];
+    [newReq setRendezvous:rendezvous];
+    
+    NSUInteger pointCount = polyline.pointCount;
+    CLLocationCoordinate2D *routeCoordinates = malloc(pointCount * sizeof(CLLocationCoordinate2D));
+    [polyline getCoordinates:routeCoordinates range:NSMakeRange(0, pointCount)];
+    NSString *pointsJSON = @"{\"points\" : [";
+    for (int c=0; c < pointCount-1; c++) {
+        NSString *this = [NSString stringWithFormat:@"[%f, %f],", routeCoordinates[c].latitude, routeCoordinates[c].longitude];
+        pointsJSON = [pointsJSON stringByAppendingString:this];
+    }
+      
+    pointsJSON= [pointsJSON stringByAppendingString:[NSString stringWithFormat:@"%@ ] }", [NSString stringWithFormat:@"[%f, %f]", routeCoordinates[pointCount-1].latitude, routeCoordinates[pointCount - 1].longitude]] ];
+    free(routeCoordinates);
+    [newReq setPolylineCoords:pointsJSON];
```

Comment:
is this coordinates to string transform done anywhere else? or do we do a reverse string to coordinates transform? maybe this can be split out into a shared utility.